### PR TITLE
options/rtdl: implement `LD_SHOW_AUXV`

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

File: options/rtdl/generic/main.cpp [hunk 1]
```diff
@@ -220,6 +236,71 @@ extern "C" void *interpreterMain(uintptr_t *entry_stack) {
 		if(!(*aux))
 			break;
 
+		if(ldShowAuxv) {
+			switch(*aux) {
+#ifdef AT_PHDR
```
Comment (64) [inline options/rtdl/generic/main.cpp:241]: These defines are really ugly, but I'm not sure there is a better way. We could perhaps try to unify the ABI headers a bit more.

**Review by 64 — CHANGES_REQUESTED** [review]

File: options/rtdl/generic/main.cpp (hunk 1 above)
Comment (ArsenArsen) [reply options/rtdl/generic/main.cpp:241]: could also replace AT_* with something like

```c
#define AT_DO_DEFINES \
        at_cb(AT_PHDR, 1) \
        at_cb(AT_PHENT, 2) \
        ...

enum {
#define at_cb(name, val) name = val,
AT_DO_DEFINES
#undef at_cb
};
```

this block would become

```c
#define at_cb(name, val) case name: mlibc::infoLogger() << #name " 0x" ...
AT_DO_DEFINES
#undef at_cb
```

Comment (no92) [reply options/rtdl/generic/main.cpp:241]: As far as I can tell, possible `a_type` values are defined in the arch ABI spec, so I don't quite see the point of having it be an abi-bit as it is right now.

Comment (avdgrinten) [reply options/rtdl/generic/main.cpp:241]: Aren't some of the AT_ constants mandatory for ld.so to work? These can be assumed to exist on all ABIs. Other `AT_` constants are Linux extensions and e.g. FreeBSD's `AT_` constants do not match Linux'.

Comment (avdgrinten) [reply options/rtdl/generic/main.cpp:241]: Also, the constants defined by the ELF ABI supplement can be assumed to exist and have values matching the standard.

**Review by 64 — APPROVED** [review]

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Review by Geertiebear — APPROVED** [review]

**Pull request merged**